src/app/EventManagement.cpp

```diff
@@ -618,9 +604,9 @@ CHIP_ERROR EventManagement::EventIterator(const TLVReader & aReader, size_t aDep
     if (event.mPriority == apEventLoadOutContext->mPriority)
     {
         apEventLoadOutContext->mCurrentSystemTime.mValue += event.mDeltaSystemTime.mValue;
-        VerifyOrExit(apEventLoadOutContext->mCurrentEventNumber < apEventLoadOutContext->mStartingEventNumber,
+        VerifyOrExit(!(apEventLoadOutContext->mCurrentEventNumber >= apEventLoadOutContext->mStartingEventNumber &&
```

Comment:
This long condition seems hard to follow especially given the negation in front. Since either of the conditions being false results in an exit, could we split into two?

```cpp
VerifyOrExit(apEventLoadOutContext->mCurrentEventNumber < apEventLoadOutContext->mStartingEventNumber, err = CHIP_EVENT_ID_FOUND)

VerifyOrExit(!IsInterestedEventPaths(apEventLoadOutContext->mpInterestedEventPaths, event), err = CHIP_EVENT_ID_FOUND);
```

Readability-wise I find this logic somewhat unclear: why do we say "I found the value" if the check is "Is this event interesting"?

Reply:
Sinc the exit branch has no cleanup part, consider using "return" instead of exit to make the code easier to follow and less macro-heavy. A `if (!IsInterested...) return CHIP_EVENT_ID_FOUND` seems easier to read than a macro expansion.

Reply:
Yep, just move this logic into IsInterested function, and update it with return pattern instead of exit.